env.py

```diff
@@ -159,9 +192,14 @@ def get_mask(self):
         else:
             return mask
 
-    def get_parents(self, backward=False):
-        if self.done:
-            if self.state[-1] == self.token_eos:
+    def get_parents(self, backward=False, state=None, done=None):
```

Comment:
Why are there two `get_parents()` methods?

Reply:
I'm a little confused which two methods are you talking about:

If you're talking about the deleted and the modified code: In the existing main-new-al implementation, ti was not possible to feed state as an input argument into the `get_parents()` function (as is required by `get_paths()`). So I modified the existing implementation to take care of that.

But like apart from that, there is an abstract method and an aptamer specific method in the current branch.